lightkurve/correctors/pldcorrector.py

```diff
@@ -82,199 +116,509 @@ class PLDCorrector(object):
-            # set up gaussian process using celerite
-            # we use a Matern-3/2 kernel for its flexibility and non-periodicity
-            kernel = celerite.terms.Matern32Term(np.log(amp), np.log(tau))
-            gp = celerite.GP(kernel)
-            gp.compute(M(self.time), M(rawflux_err))
-
-            # compute the coefficients C on the basis vectors;
-            # the PLD design matrix will be dotted with C to solve for the noise model.
-            A = np.dot(MX.T, gp.apply_inverse(MX))
-            B = np.dot(MX.T, gp.apply_inverse(M(rawflux)[:, None])[:, 0])
+            cadence_mask = np.ones(len(self.raw_lc.time), dtype=bool)
+
+        # Theano raises an `AsTensorError` ("Cannot convert to TensorType")
+        # if the floats are not in 64-bit format, so we convert them here:
+        lc_time = np.asarray(self.raw_lc.time, np.float64)
+        lc_flux = np.asarray(self.raw_lc.flux, np.float64)
+        lc_flux_err = np.asarray(self.raw_lc.flux_err, np.float64)
+
+        # Covariance matrix diagonal
+        diag = lc_flux_err**2
+
+        # The cadence mask is applied by inflating the uncertainties in the covariance matrix;
+        # this is because celerite will run much faster if it is able to predict
+        # data for cadences that have been fed into the model.
+        diag[~cadence_mask] += 1e12
+
+        with pm.Model() as model:
+            # Create a Gaussian Process to model the long-term stellar variability
+            # log(sigma) is the amplitude of variability, estimated from the raw flux scatter
+            logsigma = pm.Normal("logsigma", mu=np.log(np.std(lc_flux)), sd=4)
+            # log(rho) is the timescale of variability with a user-defined prior
+            logrho = pm.Normal("logrho", mu=np.log(gp_timescale_prior), sd=4)
+            # log(s2) is a jitter term to compensate for underestimated flux errors
+            logs2 = pm.Normal("logs2", mu=np.log(np.var(lc_flux)), sd=4)
+            kernel = xo.gp.terms.Matern32Term(log_sigma=logsigma, log_rho=logrho)
+
+            # Store the GP and cadence mask to aid debugging
+            model.gp = xo.gp.GP(kernel, lc_time, diag + tt.exp(logs2))
+            model.cadence_mask = cadence_mask
+
+            # The motion model regresses against the design matrix
+            A = tt.dot(design_matrix.T, model.gp.apply_inverse(design_matrix))
+            B = tt.dot(design_matrix.T, model.gp.apply_inverse(lc_flux[:, None]))
+            weights = tt.slinalg.solve(A, B)
+            motion_model = pm.Deterministic("motion_model", tt.dot(design_matrix, weights)[:, 0])
+            pm.Deterministic("weights", weights)
+
+            # Likelihood to optimize
+            pm.Potential("obs", model.gp.log_likelihood(lc_flux - motion_model))
+
+            # Track the flux values of the different diagnostic light curves for ease of use
+            gp_model, gp_model_var = model.gp.predict(return_var=True)
```

Comment:
The `predict` step here is pretty costly, and I don't think you're actually using it for anything. I think it would be good to try removing it, and just returning the gp object itself, or the best fit hyper parameters, and predicting only once at the end, outside of the model.